mlrun/serving/states.py

```diff
@@ -1248,6 +1248,12 @@ def params_to_step(
             )
         if not name:
             raise MLRunInvalidArgumentError("queue name must be specified")
+        # Pass full_event on only if it's explicitly defined
+        if full_event is not None:
+            new_class_args = {}
+            new_class_args.update(class_args)
+            new_class_args["full_event"] = full_event
+            class_args = new_class_args
```

Comment:
To avoid mutating the parameter for the caller, just in case.

Reply:
Got it, since you already verified that `class_args` is not `None` above, you can just do -
```
class_args = class_args.copy()
class_args["full_event"] = full_event
```